src/components/Authentication/KratosUI.tsx

```diff
@@ -55,42 +56,94 @@ const KratosButton: FC<KratosProps> = ({ node }) => {
+              {x.text}
+            </Form.Control.Feedback>
+          ))
+        : null,
+    [node]
+  );
+
   return (
     <Form.Group>
       <Form.Label>
         {getNodeTitle(node)}
-        {attributes.required && <span style={{ color: '#d93636' }}>{' *'}</span>}
+        {attributes.required && <Required />}
       </Form.Label>
-      <Form.Control
-        type={attributes.type}
-        value={state ? String(state) : ''}
-        name={getNodeName(node)}
-        onChange={e => setState(e.target.value)}
-        required={attributes.required}
-        disabled={attributes.disabled}
-      />
-      <KratosMessages messages={node?.messages} />
+      <InputGroup>
+        <Form.Control
+          type={inputType}
+          value={value ? String(value) : ''}
+          name={getNodeName(node)}
+          onChange={e => setValue(e.target.value)}
+          required={attributes.required}
+          disabled={attributes.disabled}
+          isInvalid={isInvalid}
+          autoComplete={autoComplete}
+          autoCorrect={autoCorrect}
+          autoCapitalize={autoCapitalize}
+          /*aria-labelledby={} TODO */
+        />
+        {isPassword && (
+          <InputGroup.Append>
+            <InputGroup.Text>
+              <IconButton onClick={() => setInputType(inputType === 'password' ? 'text' : 'password')}>
+                <Icon component={inputType === 'password' ? Eye : EyeSlash} color="inherit" size={'xs'} />
+              </IconButton>
+            </InputGroup.Text>
+          </InputGroup.Append>
+        )}
+        {feedbackElements}
+      </InputGroup>
     </Form.Group>
   );
 };
 
 const KratosCheckbox: FC<KratosProps> = ({ node }) => {
   const attributes = node.attributes as UiNodeInputAttributes;
-  const [state, setState] = useState(getNodeValue(node));
+  const [state, setState] = useState(Boolean(getNodeValue(node)));
+
+  const isInvalid = !!(node && Array.isArray(node.messages) && node.messages.find(x => x.type === 'error'));
+  console.log(isInvalid);
+  const feedbackElements = useMemo(
+    () =>
+      isInvalid
+        ? node.messages.map((x, key) => (
+            <Form.Control.Feedback type="invalid" key={key}>
+              {x.text}
+            </Form.Control.Feedback>
+          ))
+        : null,
+    [node]
+  );
+
   return (
     <Form.Group controlId={node.group}>
-      <Form.Check
-        type={'checkbox'}
-        label={getNodeTitle(node)}
-        value={state ? String(state) : ''}
-        onChange={e => setState(e.target.value)}
-        required={attributes.required}
-        disabled={attributes.disabled}
-      />
-      <KratosMessages messages={node?.messages} />
+      <Form.Check name={getNodeName(node)} type="checkbox">
+        <Form.Check.Input
+          type="checkbox"
+          name={getNodeName(node)}
+          checked={state}
+          onChange={() => setState(oldState => !oldState)}
+          isInvalid={isInvalid}
```

Comment:
At end `value` should be set, because the form should send value depending on the type.
```
 <Form.Check.Input
    type="checkbox"
    name={getNodeName(node)}
    value={String(state)}
    checked={state}
    onChange={() => setState(oldState => !oldState)}
    isInvalid={isInvalid}
  />
```